controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,253 @@
+	NamespacedName types.NamespacedName `yaml:"metadata"`
+	ClusterID      string               `yaml:"clusterId"`
+}
+type InfoStorageCluster struct {
+	NamespacedName          types.NamespacedName `yaml:"metadata"`
+	StorageProviderEndpoint string               `yaml:"storageProviderEndpoint"`
+	CephClusterFSID         string               `yaml:"cephClusterFSID"`
+}
+
+type OdfInfoData struct {
+	OdfVersion          string             `yaml:"odfVersion"`
+	OdfDeploymentType   string             `yaml:"odfDeploymentType"`
+	Clients             []ConnectedClient  `yaml:"clients"`
+	StorageCluster      InfoStorageCluster `yaml:"storageCluster"`
+	StorageClusterCount int                `yaml:"storageClusterCount"`
+	StorageSystemName   string             `yaml:"storageSystemName"`
+	IsDROptimized       bool               `yaml:"isDROptimized"`
+}
+
+const (
+	OdfInfoKeyName            = "config.yaml"
+	OdfDeploymentTypeExternal = "external"
+	OdfDeploymentTypeInternal = "internal"
+	RookCephMonSecretName     = "rook-ceph-mon"
+	FsidKey                   = "fsid"
+	OdfOperatorNamePrefix     = "odf-operator"
+	OdfInfoConfigMapName      = "odf-info"
+	OdfInfoMapKind            = "ConfigMap"
+)
+
+type odfInfoConfig struct{}
+
+var mutex sync.RWMutex
+
+// ensureCreated ensures that a ConfigMap resource exists with its Spec in
+// the desired state.
+func (obj *odfInfoConfig) ensureCreated(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	odfInfoConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OdfInfoConfigMapName,
+			Namespace: operatorNamespace,
+		},
+		Data: map[string]string{},
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, odfInfoConfigMap, func() error {
+		if err := ctrl.SetControllerReference(storageCluster, odfInfoConfigMap, r.Scheme); err != nil {
+			return err
+		}
+		r.Log.Info("Creating or updating odf-info configmap", OdfInfoMapKind, klog.KRef(storageCluster.Namespace, OdfInfoConfigMapName))
+		odfInfoData, configErr := getOdfInfoData(r, storageCluster)
+		if configErr != nil {
+			return fmt.Errorf("failed to get ODF info config data: %w", configErr)
+		}
+		odfInfoKey := obj.getOdfInfoKeyName(storageCluster)
+		// Creates or appends to the data map
+		odfInfoConfigMap.Data[odfInfoKey] = odfInfoData
+		return nil
+	})
+	if err != nil {
+		r.Log.Error(err, "failed to create or update odf-info config", OdfInfoMapKind, klog.KRef(storageCluster.Namespace, OdfInfoConfigMapName))
+		return reconcile.Result{}, fmt.Errorf("failed to create or update odf-info config: %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (obj *odfInfoConfig) getOdfInfoKeyName(storageCluster *ocsv1.StorageCluster) string {
+	return storageCluster.Namespace + "/" + storageCluster.Name + "." + OdfInfoKeyName
```

Comment:
For contacting multiple strings we usually use `fmt.Sprintf` instead of multiple `+` operations.
```suggestion
	return fmt.Sprintf("%s/%s.%s", storageCluster.Namespace, storageCluster.Name, OdfInfoKeyName)
```